app/listeners/iiif_print/listener.rb

```diff
@@ -29,5 +29,20 @@ def on_file_set_attached(event, service: IiifPrint::SplitPdfs::ChildWorkCreation
 
       service.conditionally_enqueue(file_set: file_set, work: work, file: file, user: user)
     end
+
+    ##
+    # Responsible for setting the is_child flag on the work when a child work is created.
+    #
+    # @param event [#[]] a hash like construct with :object key
+    def on_object_deposited(event)
```

Comment:
I thought we had talked about not using a listener but instead adding a step at the end of `change_set.update_work`?  

The [`on_object_deposited` appears 8 times in the graph of events and listeners](https://gist.github.com/jeremyf/01d00ed58d9cf6f8237851fcc4a817a0); when creating/updating AdminSet/Collection/Work/FileSet.  Also, the `object.deposited` fires before we attach members, which might mean we call this code before we have the children assigned.

Looking at [the graph of events for `change_set.update_work`](https://gist.github.com/jeremyf/01d00ed58d9cf6f8237851fcc4a817a0#file-graph-yml-L247-L287) we likely want to add child works as a step after `Hyrax:: Transactions:: Steps: :UpdateWorkMembers`

<img width="487" alt="Screenshot 2024-01-23 at 8 55 27 AM" src="https://github.com/scientist-softserv/iiif_print/assets/2130/525d1b85-0fe8-4e82-9b9a-b9cdc3c34bcc">

With the current implementation the `event[:object]` could be a new AdminSet, Collection, FileSet, or Work.  I would assume that if this is a Collection, we'd find all of that collection's works and stamp the `is_child` on it.

Reply:
I did not recall if we decided to use a listener or transaction but vaguely remember leaning towards one way but couldn't find what that was.  When I was going through the Transitioning to Valkyrie workshop, it seemed like the listeners were appropriate concept to replace callbacks.

As for getting called multiple times, can we do a check for if it's a curation concern?  If not, bail

If we're set on transactions then we can just close this and I'll pursue that route.